dbms/src/IO/S3Common.cpp

```diff
@@ -106,46 +105,37 @@ namespace S3
 
     URI::URI(const Poco::URI & uri_)
     {
-        static const std::regex bucket_key_pattern("([^/]+)/(.*)"); /// TODO std::regex is discouraged
+        static const RE2 virtual_hosted_style_pattern("(.+\\.)?s3[.-][a-z0-9-.]+");
```

Comment:
Maybe you can provide a reference in comments that will answer the question "what is that"?

Reply:
Don't understand:

`[.-]` means "from dot symbol to the end of byte range", is that correct?

Reply:
No, it means that there shouldbe "." or "-" after "s3"

Reply:
Ok, I've added another backslash for convenience.